homeassistant/components/modern_forms/__init__.py

```diff
@@ -53,9 +55,60 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Set up all platforms for this device/entry.
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
+    async_load_modern_forms_device_services(hass)
     return True
 
 
+@callback
+def async_load_modern_forms_device_services(hass: HomeAssistant) -> None:
+    """Set up services for the Modern Forms integration."""
+    if hass.services.has_service(DOMAIN, SERVICE_REBOOT):
+        # Integration-level services have already been added. Return.
+        return
+
+    async def extract_modern_forms_config_entry_ids(
+        service_call: ServiceCall,
+    ) -> list[str]:
+        entries: list[str] = []
+        for entry_id in await async_extract_config_entry_ids(hass, service_call):
+            config_entry: ConfigEntry | None = hass.config_entries.async_get_entry(
+                entry_id
+            )
+            if config_entry and config_entry.domain == DOMAIN:
+                entries.append(entry_id)
+        return entries
+
+    async def async_reboot(service_call: ServiceCall) -> None:
+        """Reboot the Modern Forms fan control unit."""
+        if not (
+            modern_forms_entry_ids := await extract_modern_forms_config_entry_ids(
+                service_call
+            )
+        ):
+            raise HomeAssistantError(
+                f"Failed to call service '{SERVICE_REBOOT}'. Config entry for target not found"
+            )
+        for entry_id in modern_forms_entry_ids:
+            coordinator = hass.data[DOMAIN][entry_id]
+            await coordinator.modern_forms.reboot()
+
+    hass.services.async_register(DOMAIN, SERVICE_REBOOT, async_reboot, None)
+
+
+@callback
+async def async_unload_modern_forms_services(hass: HomeAssistant) -> None:
+    """Unload services for the Modern Forms fan integration."""
+    if DOMAIN in hass.data and hass.data[DOMAIN]:
```

Comment:
Why the second test ?

Reply:
The first checks if it has the key in the dict.  The second checks if it has content.  Basically to check the check it without raising a key missing error

Reply:
I see that but can you ever have None as content, so far I have not seen that.